nilearn/datasets/func.py

```diff
@@ -206,6 +143,13 @@ def fetch_haxby(data_dir=None, n_subjects=1, fetch_stimuli=False,
             **kwargs)
 
 
+@deprecated("fetch_haxby_simple will be removed in future releases. "
+            "Calling 'fetch_haxby' instead.")
```

Comment:
Could you also please remove it from doc/modules/reference.rst.

Aside from that, +1 for merge from my side, once CI has run.